celery/backends/s3.py

```diff
@@ -48,9 +50,13 @@ def __init__(self, **kwargs):
 
         self.base_path = conf.get('s3_base_path', None)
 
+        self.partition_by_date = conf.get('s3_partition_by_date', False)
+
         self._s3_resource = self._connect_to_s3()
 
     def _get_s3_object(self, key):
+        if self.partition_by_date:
```

Comment:
can you please add proper tests to the proposed change? it is hard to verify for us without having any test cases. also can you please point to the S3 docs relevant to the change?